packages/vela/src/Vela.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  type MintActionParams,
+  type TransactionFilter,
+  type SwapActionParams,
+  type StakeActionParams,
+  compressJson,
+} from '@rabbitholegg/questdk'
+import { type Address } from 'viem'
+import { VAULT_CONTRACT, CHAIN_TO_TOKENS, TOKENFARM_CONTRACT, VLP_CONTRACT, VELA_CONTRACT } from './contract-addresses'
+import { CHAIN_ID_ARRAY } from './chain-ids'
+import { TOKENFARM_ABI, TOKENFARM_ABI2, VAULT_ABI } from './abi'
+
+export const swap = async (
```

Comment:
You might just want to take this out and use our new derivatives action type instead of a swap. It would fit this action much better and have much better support on the front end once it (derivatives action) has been fully implemented.

Reply:
@mmackz 
I checked your derivatives action type, but I dont think it is not fit with our new position orderPacked.
it includes only two bigInt parameters, but we need three bigInt parameters.